### PR TITLE
#21889 Change Vanity URL field type to HostFolderField

### LINKED ISSUE #21889: Create a startup task to change Vanity URL "site" field to be a host folder field
For all Content types with base type `VANITY_URL` we need to update the `site` field, which is a custom_field that calls a vtl to a `com.dotcms.contenttype.model.field.HostFolderField`

We will then need to change where we look up the vanity url `site` in code.   

There might be specific logic in the content checkin method for vanity urls that will will need to remove.  I know there is validation that is required here that needs to change.

https://github.com/dotCMS/core/blob/92ece95ec3fe6223d9d74ec756734195229ee558/dotCMS/src/main/java/com/dotcms/vanityurl/business/VanityUrlAPIImpl.java#L267

### PR DESCRIPTION
**Pull request by freddyucv**
Changing the Vanity URL's site field type to HostFolderField.
Also create a Task Upgrade to change the field type in DataBase
Also avoid to use this: `vanityURL.getStringProperty(VanityUrlContentType.SITE_FIELD_VAR` to get the Vanity URL site

### CODE REVIEW

File: dotCMS/src/main/java/com/dotmarketing/startup/runonce/Task220330ChangeVanityURLSiteFieldType.java [hunk 1]
```diff
@@ -0,0 +1,42 @@
+package com.dotmarketing.startup.runonce;
+
+import com.dotmarketing.startup.AbstractJDBCStartupTask;
+import java.util.List;
+
+public class Task220330ChangeVanityURLSiteFieldType extends AbstractJDBCStartupTask  {
+
+    private String getScript(){
+        return "UPDATE field SET field_type = 'com.dotcms.contenttype.model.field.HostFolderField' "
+                + "WHERE field_name = 'Site' and structure_inode in (select inode from structure where structuretype = 7)";
```
Comment (wezell) [inline dotCMS/src/main/java/com/dotmarketing/startup/runonce/Task220330ChangeVanityURLSiteFieldType.java:10]: where clause should use the `velocity_var_name` as field_name is not unique

**Review by wezell — CHANGES_REQUESTED** [review]

**Review by jdotcms — APPROVED** [review]

File: dotCMS/src/main/java/com/dotmarketing/startup/runonce/Task220330ChangeVanityURLSiteFieldType.java [hunk 2]
```diff
@@ -0,0 +1,149 @@
...
+import com.dotmarketing.startup.StartupTask;
+import com.dotmarketing.util.Logger;
+import com.dotmarketing.util.UtilMethods;
+import com.fasterxml.jackson.core.JsonProcessingException;
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Map;
+import java.util.stream.Collectors;
+
+public class Task220330ChangeVanityURLSiteFieldType implements StartupTask {
```
Comment (victoralfaro-dotcms) [inline dotCMS/src/main/java/com/dotmarketing/startup/runonce/Task220330ChangeVanityURLSiteFieldType.java:21]: Javadoc

Comment (dsilvam) [inline dotCMS/src/main/java/com/dotmarketing/startup/runonce/Task220330ChangeVanityURLSiteFieldType.java:21]: javadoc

**Comment by wezell** [comment]
Guys, can we get this merged so we can cherry-pick it into a 5.3.8 test branch?  We can do the javadoc after this.  A customer is waiting for a fix.

**Review by dsilvam — APPROVED** [review]

File: dotCMS/src/main/java/com/dotmarketing/startup/runonce/Task220330ChangeVanityURLSiteFieldType.java (hunk 2 above)
Comment (freddyucv) [reply dotCMS/src/main/java/com/dotmarketing/startup/runonce/Task220330ChangeVanityURLSiteFieldType.java:36]: done https://github.com/dotCMS/core/pull/21923/commits/a0f04ac75eadcd9317cbf91cb0039ca074f80011

Comment (freddyucv) [reply dotCMS/src/main/java/com/dotmarketing/startup/runonce/Task220330ChangeVanityURLSiteFieldType.java:36]: done https://github.com/dotCMS/core/pull/21923/commits/a0f04ac75eadcd9317cbf91cb0039ca074f80011

**Review by victoralfaro-dotcms — APPROVED** [review]
LGTM

**Review by dsilvam — APPROVED** [review]

**Pull request merged**